WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -466,6 +474,37 @@ class BattleView extends React.Component {
         }
     }
 
+    visualizeEffects = (effects) => {
+        for (let effect of effects) {
+            switch (effect.id) {
+                case 1:
+                    this.visualizeDamageEffect(effect.target_player);
+                    break;
+                default:
+                    console.log("unidentified effect");
+                    break;
+            }
+        }
+    }
+
+    visualizeDamageEffect = (target) => {
+        console.log(`damage effect: ${target}, user: ${this.state.user}`);
+        if (target === this.state.user) {
+            this.setState({
+                damageBorderOpacity: {user: '0.5', enemy: '0'}
+            });
+        } else {
+            this.setState({
+                damageBorderOpacity: {user: '0', enemy: '0.5'}
+            });
+        }
+        setTimeout(() => {
+            this.setState({
+                damageBorderOpacity: {user: '0', enemy: '0'}
+            });
+        }, nextStepAnimationDuration);
+    }
+

```

Comment:
Taki sposób nie działa: mam np. efekt zadawanie dmg jako ostatni, jako pierwszy efekto randomizacji decka. Przy pierwszym efekcie, który jest efektem losowanie kart to powinno nie być żadnej wizualizacji, ale wykonuje się wizualizacja damage. 

To rozwiązanie sprawia, że wykonują się wszystkie wizualizacje efektów naraz za każdym razem

Reply:
Kurde źle zaznaczyłem kod, sorry, chodziło mi o funkcję `visualizeEffects`